### ISSUE TITLE
find optimal Windows installer package

### ISSUE THREAD

**Issue by bruhwile** [radiasoft/radtrack#18]

**Comment by robnagler** [comment]
I thought I had a solution which was "easy", but it is failing with:

```
_msi.MSIError: Could not insert [('__init__.py.591', 'scripts4', '__INIT__.PY|__init__.py', 0L, None, None, 512, 32768)] into File
```

The `32768` looked suspect, which led me to the fact that the limit is 32,767 on the number of installed files: https://msdn.microsoft.com/en-us/library/aa367767%28v=vs.85%29.aspx 

What I was trying to do was include all of Anaconda so I wouldn't have the dependency problem. I got 
the python startup to work with cx_freeze, but then this happened. These "builds" take about an hour
on my Mac running a Windows 8 VM so it is pretty slow going.

We may have to have the installer pull down other installers (miktex and anaconda) dynamically, and run them behind the scenes. My first attempt tomorrow will be to create a custom action that will unzip blobs.

**Comment by robnagler** [comment]
David, you can find a RadTrack win32 installer:

bruhwile@dev.radiasoft.org:RadTrack-1.0-win32.msi (about 1GB)

The first phase of the install is just like normal. You select what directory you want, and it puts files there. It puts an icon on your desktop.

When you run RT the first time, it will ask you again for permissions to modify things, and you'll get a console window telling you it's doing a second install phase. This one takes more time than the initial install, because it is unpacking Anaconda, TeX, and APS. Once the unpacking is complete, it will start RadTrack.

There will be an extra console window hanging around while RT runs. I will probably drop that once I've debugged the process. If you get an error during the install or running the program, you'll see it in the console window. It may go away quickly.

I didn't try this on Win7 yet. Just try it on Win8 if that's convenient, and I'll debug Win7 once I know it works for you for Win8.

**Comment by bruhwile** [comment]
Thanks a lot, Rob.

At first glance, the install on Win8 was fine.
Only exception is a lingering windows terminal that I had to close by hand.

The lower plots on the beam tab are being clipped.

Which executables are included?
I'll try to run something with Elegant, but need to do other things for a
couple of hours.

David

On Tue, Feb 24, 2015 at 6:08 PM, Rob Nagler <notifications@github.com>
wrote:

> David, you can find a RadTrack win32 installer:
>
> bruhwile@dev.radiasoft.org:RadTrack-1.0-win32.msi (about 1GB)
>
> The first phase of the install is just like normal. You select what
> directory you want, and it puts files there. It puts an icon on your
> desktop.
>
> When you run RT the first time, it will ask you again for permissions to
> modify things, and you'll get a console window telling you it's doing a
> second install phase. This one takes more time than the initial install,
> because it is unpacking Anaconda, TeX, and APS. Once the unpacking is
> complete, it will start RadTrack.
>
> There will be an extra console window hanging around while RT runs. I will
> probably drop that once I've debugged the process. If you get an error
> during the install or running the program, you'll see it in the console
> window. It may go away quickly.
>
> I didn't try this on Win7 yet. Just try it on Win8 if that's convenient,
> and I'll debug Win7 once I know it works for you for Win8.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/radiasoft/radtrack/issues/18#issuecomment-75885944>.
>



-- 
David Bruhwiler
RadiaSoft LLC
+1 (720) 502-3928

**Comment by robnagler** [comment]
I just tried to drop the terminal window by setting up the app as a Win32GUI. This works, but it doesn't show the console window with the unzips. Since there's no locking on the unzips, it would cause thrashing if someone clicked multiple times without knowing that the unzips are happening in background.

I can fix `radtrack_start_win32.py` to invoke `RbGlobal.py` in background. Another alternative is to make the second phase installer a PyQt app with locking (a good thing in any event). 

For now, I think we should leave it with the lingering console window so Bryce can get it going.

Here's a public link for the previous installer: [[https://user.radiasoft.org/nagler/pub/RadTrack-20150224.msi]]

**Comment by bruhwile** [comment]
Thanks a lot, Rob!

I'll close the issue.

If more work is required on the installer, we'll open a new issue.

Mark -- if you have a chance, please give the installer a try.

Thanks,
David

**Comment by MZHarrison** [comment]
The installation was successful on my computer (Windows 7).

Why is the installation a two-stage operation? Why not do all the unpacking during the installation instead of putting a 10-minute delay in the first run of RadTrack? This would also solve the issue:
> Since there's no locking on the unzips, it would cause thrashing if someone clicked multiple times without knowing that the unzips are happening in background.

**Comment by MZHarrison** [comment]
The uninstaller doesn't remove the Anaconda, APS, or tex folders.

**Comment by robnagler** [comment]
The uninstaller doesn't remove Anaconda, APS, and tex folders. The next
install will delete them before installing the new versions.

The problem with the installer is that it relies on cx_freeze, which is a
quick way of building an installer for a Python app. It was a "quick hack"
to get Bryce using RT with all the necessary dependencies.

**Comment by bruhwile** [comment]
Closing again. The installer is good enough for now. The limiting factors are elsewhere. We will open a new issue or issues in the future, when it is time to improve the installer.
